### PR TITLE
Honeycomb Integration

### PR DESCRIPTION
**Pull request by pjdufour-truss**
## Description

This pull request adds [Honeycomb](https://www.honeycomb.io/) instrumentation to MyMove.  Honeycomb integration is disabled by default and requires 3 environment variables to be set.  At this stage, we'd only set these variables in staging and experimental.

## Reviewer Notes

I didn't fully test the application with the update to the dependencies, but I guess that's why we have end to end tests!?  The application would start properly with and without the environment variables.

## Setup

We'll need to add environment variables for `honeycomb_enabled`, `honeycomb_api_key `, and `honeycomb_dataset` to our task definitions for staging and experimental.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

### CODE REVIEW

**Comment by dynamike** [comment]
@ntwyman Would you also mind taking a look? The plan is to test out using Honeycomb in experimental and staging only to road test it's value as a tool for debugging. The beeline-go library integrates with goji to pull out useful fields about HTTP requests.

**Comment by pjdufour-truss** [comment]
`honeycomb_api_key` is actually a secret so that'll be managed with chamber.

File: cmd/webserver/main.go [hunk 1]
```diff
@@ -307,17 +336,30 @@ func main() {
 	go func() { // start http listener
 		addr := fmt.Sprintf("%s:%s", *listenInterface, *port)
 		zap.L().Info("Starting http server listening", zap.String("address", addr))
-		s := &http.Server{
-			Addr:           addr,
-			Handler:        site,
-			MaxHeaderBytes: maxHeaderSize,
+		if useHoneycomb {
```
Comment (ntwyman) [inline cmd/webserver/main.go]: I would not do this down here but at line 335
```
	errChan := make(chan error)
    if useHoneycomb {
        site = hnynethttp.WrapHandler(site)
    }
	go func() { // start http listener

Comment (ntwyman) [inline cmd/webserver/main.go]: Also - why does honeycomb need to wrap the site as well as the middleware on the API at line 286?

Comment (pjdufour-truss) [reply cmd/webserver/main.go]: It's honeycomb's way of deciding which routes are captured.  We only want to capture `/api`.

Comment (ntwyman) [reply cmd/webserver/main.go]: That way you dont need to wrap all the server constructors and ListenAndServe calls in if useHoneyComb calls

Comment (pjdufour-truss) [reply cmd/webserver/main.go]: I was just following the docs at https://github.com/honeycombio/beeline-go/blob/master/examples/goji/main.go.  I would not assume the `wrapper` would implement all the correct functions as called in the lines in between.

Comment (ntwyman) [reply cmd/webserver/main.go]: Thumbs up

Comment (ntwyman) [reply cmd/webserver/main.go]: There is no calling inbetween 335 is right  before the firs co-routine

**Pull request merged**